modules/packages/LinearAlgebra.chpl

```diff
@@ -1955,6 +1997,21 @@ module Sparse {
 
     return B;
   }
+  
+  proc _array.elementDiv(A: [?Adom] ?eltType) where 
+                                            isSparseArr(this) && !isCSArr(this)
+                                            && isSparseArr(A) && !isCSArr(A) {
+    if Adom.rank != this.domain.rank then compilerError("Unmatched ranks");
+    if this.domain.shape != Adom.shape then halt("Unmatched shapes");
+    var sps: sparse subdomain(Adom.parentDom);
+    sps += this.domain;
+    sps += Adom;
+    var S: [sps] eltType;
+    forall (i,j) in sps {
+      S[i,j] = this[i,j] / A[i,j];
```

Comment:
This is where you should be careful. `sps` contains indices from both arrays. And if we are implementing a generic case, it is likely that for some of those indices A[i,j] will have zero values. I think it is ok to keep resulting matrix's domain as union of both, but the actual elementwise operation should be done on A.domain and not sps.

Reply:
Right, I was considering cases of 0-division but somehow overlooked this.